frontends/infinite-corridor/src/pages/SearchPage.tsx

```diff
@@ -32,94 +24,43 @@ const facetMap: FacetManifest = [
   ["offered_by", "Offered By"]
 ]
 
-interface Result {
-  _source: LearningResourceSearchResult
-}
-
-const SEARCH_API_URL = "search/"
-
-const search = async (params: SearchQueryParams) => {
-  const body = buildSearchQuery(params)
-  try {
-    const { data } = await axios.post(SEARCH_API_URL, body)
-    return data
-  } catch (err) {
-    return null
-  }
-}
-
 const SearchPage: React.FC = () => {
-  const [results, setSearchResults] = useState<Result[]>([])
-  const [total, setTotal] = useState(0)
-  const [completedInitialLoad, setCompletedInitialLoad] = useState(false)
-  const [requestInFlight, setRequestInFlight] = useState(false)
-  const [searchApiFailed, setSearchApiFailed] = useState(false)
-  const [aggregations, setAggregations] = useState<Aggregations>(new Map())
   const isMd = useMuiBreakpoint("md")
 
-  const clearSearch = useCallback(() => {
-    setSearchResults([])
-    setCompletedInitialLoad(false)
-    setTotal(0)
-  }, [])
-
-  const runSearch = useCallback(
-    async (text: string, activeFacets: Facets, from: number) => {
-      if (activeFacets["type"]) {
-        activeFacets["type"] = intersection(ALLOWED_TYPES, activeFacets["type"])
-      } else {
-        activeFacets["type"] = ALLOWED_TYPES
-      }
-      setRequestInFlight(true)
-      const newResults = await search({
-        text,
-        from,
-        activeFacets,
-        size: pageSize
-      })
-      setRequestInFlight(false)
-
-      if (!newResults || newResults["apiFailed"]) {
-        setSearchApiFailed(true)
-        return
-      }
+  const history = useHistory()
+  const search = useSearchInputs(history)
+  const { text, activeFacets } = search.searchParams
+  const searchQuery = useInfiniteSearch({
+    text,
+    activeFacets,
+    size:         pageSize,
+    allowedTypes: ALLOWED_TYPES
+  })
+  useSyncUrlAndSearch(history, search)
 
-      setAggregations(new Map(Object.entries(newResults.aggregations ?? {})))
+  const aggregations = useMemo(() => {
+    return new Map(
+      Object.entries(searchQuery.data?.pages[0].aggregations ?? {})
```

Comment:
**Question: ** @abeglova if two queries have same facets / text, just different `from` parameters, then they have the same `aggregations`, right? I'm just using the first page's aggregations.